addons/l10n_pk_edi/models/product_product.py

```diff
@@ -0,0 +1,57 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    l10n_pk_edi_sale_type = fields.Many2one('l10n.pk.product.type', 'Sale Type',
+        compute='_compute_l10n_pk_edi_sale_type', required=True, readonly=False, store=True,
+    )
+    l10n_pk_edi_schedule_code = fields.Many2one('l10n.pk.schedule.code', 'Schedule Code',
+        compute='_compute_l10n_pk_edi_schedule_code', required=True, readonly=False, store=True,
+    )
+
+    @api.depends('type')
+    def _compute_l10n_pk_edi_sale_type(self):
+        for line in self:
+            code_map = {'consu': 'T1000017', 'service': 'T1000018'}
+            code = code_map.get(line.type, 'T1000083')
```

Comment:
```suggestion
        code_map = {'consu': 'T1000017', 'service': 'T1000018'}
        for line in self:
            code = code_map.get(line.type, 'T1000083')
```
@masm-odoo can you define `T1000083` I hope you remember by last remarks 😄 
also make a small comment so other dev can understand it.

Reply:
If the line's type is not found in the code_map, use `T1000083` as a default code. Where `T1000083` is code for `Other`